src/SourceFile.js

```diff
@@ -20,14 +20,22 @@ export class SourceFile {
   encodedLiterals = new Map();
   warnings = [];
 
-  constructor(sourceCodeString) {
+  constructor(sourceCodeString, customProbes = [], mergeMode = "append") {
```

Comment:
I have created an `astOptions` object containing `customProbes`and `isReplacing` : 

```
astOptions: {
  customProbes:[],
  isReplacing: false
}
```

It can be extended later if needed